fdbclient/NativeAPI.actor.cpp

```diff
@@ -583,6 +583,14 @@ ACTOR static Future<Void> monitorClientInfo( Reference<AsyncVar<Optional<Cluster
 	}
 }
 
+void DatabaseContext::initialize( Reference<ClusterConnectionFile> connFile, int apiVersion, DatabaseContext *db ) {
+	Reference<Cluster> cluster(new Cluster(connFile, apiVersion));
+	Reference<AsyncVar<ClientDBInfo>> clientInfo(new AsyncVar<ClientDBInfo>());
+	Future<Void> clientInfoMonitor = monitorClientInfo(cluster->getClusterInterface(), connFile, clientInfo);
+
+	new (db) DatabaseContext( cluster, clientInfo, clientInfoMonitor, LiteralStringRef(""), TaskDefaultEndpoint, LocalityData(), true, false, apiVersion );
+}
```

Comment:
Hm, it's a little unfortunate that the logic here and in `create` are the same. Not sure if there is an easy way to consolidate other than maybe allocating a `DatabaseContext` pointer in `create` and then calling `initialize` before wrapping it in a `Database`.

Reply:
I'm not sure if I'd like preallocating the `DatabaseContext` in `create`, but we could potentially change `initialize` to return the pointer and allocate it using regular new if the passed in pointer is null.

Reply:
Yeah, I think that sounds fine, too.